src/commands/apphosting-config-export.ts

```diff
@@ -26,28 +30,36 @@ export const command = new Command("apphosting:config:export")
     const currentDir = process.cwd();
 
     // Get all apphosting yaml files ignoring the apphosting.local.yaml file
-    const yamlFilePaths = allYamlPaths(currentDir)?.filter(
-      (path) => !path.endsWith(APPHOSTING_LOCAL_YAML),
+    const yamlFilePaths = discoverConfigsInProject(currentDir)?.filter(
+      (path) => !path.endsWith(APPHOSTING_LOCAL_YAML_FILE),
     );
 
     if (!yamlFilePaths) {
-      logger.warn("No apphosting YAMLS found");
+      logger.warn("No apphosting.yaml found");
       return;
     }
 
     // Load apphosting.local.yaml file if it exists. Secrets should be added to the env list in this object and written back to the apphosting.local.yaml
-    const apphostingLocalConfigPath = yamlPath(currentDir, APPHOSTING_LOCAL_YAML);
-    const localAppHostingConfig = await loadAppHostingYaml(apphostingLocalConfigPath ?? undefined);
+    const localApphostingConfigPath = yamlPath(currentDir, APPHOSTING_LOCAL_YAML_FILE);
+    let localAppHostingConfig: AppHostingYamlConfig = AppHostingYamlConfig.empty();
 
-    const configsToExport = await getConfigToExport(yamlFilePaths);
-    const secretsToExport = configsToExport.secrets;
+    if (localApphostingConfigPath) {
+      localAppHostingConfig = await AppHostingYamlConfig.loadFromFile(localApphostingConfigPath);
+    }
+
+    const configToExport = await getConfigToExport(
+      yamlFilePaths,
+      options.secrets as string | undefined,
+    );
```

Comment:
super small nit: Is there a way to make this more self documenting?

getConfigToExport(yamlFilePaths, options.secrets) 

doesn't really mean anything to me. Alternatively, this means a lot

loadConfigForEnvironment("staging")

I'd recommend taking a pass through this method and seeing if we can make some of this easier to parse at first glance